src/litdata/utilities/train_test_split.py

```diff
@@ -0,0 +1,34 @@
+from copy import deepcopy
+from typing import List
+
+from litdata import StreamingDataset
+
+
+def train_test_split(streaming_dataset: StreamingDataset, splits: List[float]) -> List[StreamingDataset]:
+    """Split a StreamingDataset into multiple subsets for purposes such as training, testing, and validation.
+
+    Arguments:
+        streaming_dataset (StreamingDataset): An instance of StreamingDataset that needs to be split.
+        splits (List[float]): List of floats representing the proportion of data to be allocated to each split
+                            (e.g., [0.8, 0.1, 0.1] for train, test, and validation).
+
+    Returns:
+        List[StreamingDataset]: A list of StreamingDataset instances, each corresponding to the proportions specified
+                                in the splits argument.
+
+    """
+    if any([not isinstance(split, float) for split in splits]) or not (
```

Comment:
Let's split the ifs, so we can provide better error message to the users.

Reply:
done.